src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -143,32 +145,40 @@ internal class CSharpCodeCleanupService : ICodeCleanupService
             return document;
         }
 
-        private async Task<Document> ApplyCodeFixesAsync(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
+        private async Task<Document> ApplyCodeFixesAsync(
+            Document originalDoc, DocumentOptionSet docOptions, CancellationToken cancellationToken)
         {
-            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+            var fixAllService = originalDoc.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
 
-            var dummy = new ProgressTracker();
-            foreach (var diagnosticId in GetEnabledDiagnosticIds(docOptions))
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                var textSpan = new TextSpan(0, syntaxTree.Length);
+            var tracker = new ProgressTracker();
+            var diagnosticIds = GetEnabledDiagnosticIds(docOptions).ToImmutableArray();
```

Comment:
>GetEnabledDiagnosticIds(docOptions).ToImmutableArray() [](start = 32, length = 54)

Can we change `GetEnabledDiagnosticIds` to return an immutable array and change it's implementation to use an `ArrayBuilder` instead of a `List`?

Reply:
Sure!